contracts/StakerV2.sol

```diff
@@ -0,0 +1,948 @@
+      uint256 _pendingPoints = getPendingPoints(poolToken, _user);
+      pendingTotal = pendingTotal + _pendingPoints;
+    }
+    uint256 spentTotal = userSpentPoints[_user];
+    return (concreteTotal + pendingTotal) - spentTotal;
+  }
+
+  /**
+    Return the total number of points that the user has ever accrued.
+    @return the total number of points that the user has ever accrued.
+  */
+  function getTotalPoints(address _user) external view returns (uint256) {
+
+    uint256 concreteTotal = userPoints[_user];
+    uint256 pendingTotal = 0;
+    for (uint256 i = 0; i < poolTokens.length; ++i) {
+      IERC20 poolToken = poolTokens[i];
+      uint256 _pendingPoints = getPendingPoints(poolToken, _user);
+      pendingTotal = pendingTotal + _pendingPoints;
+    }
+    return concreteTotal + pendingTotal;
+  }
+
+  /**
+    Return the total number of points that the user has ever spent.
+    @return the total number of points that the user has ever spent.
+  */
+  function getSpentPoints(address _user) external view returns (uint256) {
+    return userSpentPoints[_user];
+  }
+
+   /**
+    Update the pool corresponding to the specified token address.
+    @param _token the address of the asset to update the corresponding pool for.
+  */
+  function updatePool(IERC20 _token) internal {
+
+    PoolInfo storage pool = poolInfo[_token];
+    if (block.timestamp <= pool.lastRewardEvent) {
+      return;
+    }
+    uint256 poolTokenSupply = pool.token.balanceOf(address(this));
+    if (poolTokenSupply <= 0) {
+      pool.lastRewardEvent = block.timestamp;
+      return;
+    }
+
+    // Calculate token and point rewards for this pool.
+    uint256 totalEmittedTokens = getTotalEmittedTokens(pool.lastRewardEvent, block.timestamp);
+    uint256 tokensReward = ((totalEmittedTokens * pool.tokenStrength) / totalTokenStrength) * 1e12;
+    uint256 totalEmittedPoints = getTotalEmittedPoints(pool.lastRewardEvent, block.timestamp);
+    uint256 pointsReward = ((totalEmittedPoints * pool.pointStrength) / totalPointStrength) * 1e30;
+
+    // Directly pay developers their corresponding share of tokens and points.
+    for (uint256 i = 0; i < developerAddresses.length; ++i) {
+      address developer = developerAddresses[i];
+      uint256 share = developerShares[developer];
+      uint256 devTokens = (tokensReward * share) / 100000;
+      tokensReward = tokensReward - devTokens;
+      uint256 devPoints = (pointsReward * share) / 100000;
+      pointsReward = pointsReward - devPoints;
+      token.safeTransfer(developer, devTokens / 1e12);
+      userPoints[developer] = userPoints[developer] + (devPoints / 1e30);
+    }
+
+    // Update the pool rewards per share to pay users the amount remaining.
+    pool.tokensPerShare = pool.tokensPerShare + (tokensReward / poolInfo[_token].tokenBoostedDeposit);
+    pool.pointsPerShare = pool.pointsPerShare + (pointsReward / poolInfo[_token].pointBoostedDeposit);
+    pool.lastRewardEvent = block.timestamp;
+  }
+
```

Comment:
missing comment header here.  also private vs internal not sure if makes a difference